### PR TITLE
[FEA-495] hub product code samples

### PR DESCRIPTION
**Pull request by fliptopbox**
*Hub products that require JsBin assets.*

Please note:
- These samples are based on an existing .code example, where the javascript portion was replaced with the respective code snippet
- The HTML section will need to be reviewed we may require copy updates.
- ANY script change will need to be communicated upstream, so that the website UI matches the JSBin content. We may need to merge this branch early (to QA the TRY IT button has the correct URL) WDYT?
- The snippet tokens (aka API_KEY and FULLY_QUALIFIED_CHANNEL_NAME) will not be dynamically replaced. The *channel name* (if it is present) is replaced at runtime, and our JSBin assets require pre-rendered assets. (ask @fliptopbox or @kennethkalmer for details) 

*These assets will not be available to the website until they have been published.*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,26 +28,117 @@ The website consumes this repo through a ruby gem. The gem points at the `master
...
+editors can simply skip to [Forking and running locally](#forking-and-running-locally).
 
 If you will be creating or modifying code within this repository that is to be uploaded automatically
-to [JSBin](https://jsbin.ably.io/), for our "Try it now code editor",
-then you will need to create yourself a JSBin config file:
+to [JSBin](https://jsbin.ably.io/), for our "Try it now code editor", then you will need to create yourself a JSBin config file:
+
+### Ably staff members
+
+To obtain the API key you need to run this command in the terminal, you must ensure you have installed the hero-cli tools (Instructions on [installing the Heroku toolbelt](https://devcenter.heroku.com/articles/heroku-cli) )
```
Comment (MarkWoulfeAbly) [inline README.md:43]: @fliptopbox you also need to have postgresql installed, which i did not

Comment (kennethkalmer) [reply README.md:43]: `hero-cli` or `heroku-cli` or `heroku toolbelt`

Comment (fliptopbox) [reply README.md:43]: @MarkWoulfeAbly I will add a mention to postgres install guide
@kennethkalmer not sure what your mean. please explain

File: README.md [hunk 2]
```diff
@@ -22,23 +22,121 @@ At this point you can either run the repo using Docker or directly with Ruby.
...
 
+## Publishing to JSBin
+
+This configuration is not required for basic documentation creation and modification, so most
+editors can simply skip to [Forking and running locally](#forking-and-running-locally).
+
+If you will be creating or modifying code within this repository that is to be uploaded automatically
+to [JSBin](https://jsbin.ably.io/), for our "Try it now code editor", then you will need to create yourself a JSBin config file:
+
+### Ably staff members
```
Comment (MarkWoulfeAbly) [inline README.md:41]: I don't see a section for non-staff members. While not a priority to get this working for non-staff members we shouldn't leave open ended headings like this.

File: README.md [hunk 3]
```diff
@@ -22,23 +22,121 @@ At this point you can either run the repo using Docker or directly with Ruby.
...
+```bash
+$ cp config/jsbin_config.example.yaml config/jsbin_config.yaml
+```
+
+### Compile the JSBin code snippets
+
+Once you have finished creating your code snippet you will need to add the new assets to the existing ones. This is essentially a table of hashed documents and corresponding file names, [./jsbins.yaml](./jsbins.yaml) if you are interested in the entire list.
+
+```bash
+$ bundle exec nanoc compile
```
Comment (MarkWoulfeAbly) [inline README.md:87]: Can we also give the docker version of this? I just re-ran docker-compile up to do this.

File: README.md [hunk 4]
```diff
@@ -22,23 +22,121 @@ At this point you can either run the repo using Docker or directly with Ruby.
...
+    update  [1.10s]  output/sse/index.html
+    update  [1.02s]  output/concepts/socketio/index.html
+```
+
+This process has added a record to `jsbin.yaml` Please notice the `ID` string in the Published URL (above) which is`ujehow` By hashing the contents, we don't need to contact the JSBin API on every static build. We only do this if a hash does not exist, and thus we need to notify JSBin to create a JSBin which contains the HTML, CSS & JS.
+
+```yaml
+---
+jsbin_hash:
+	Rh81oraUHHJ3PrrvwdJAOrVBqA8=: ujehow <-- hash: ID
```
Comment (MarkWoulfeAbly) [inline README.md:105]: The red blocks here and below are a little odd, can we remove them?

File: README.md (hunk 2 above)
Comment (fliptopbox) [reply README.md:41]: Non-staff can't update JSBins

File: README.md (hunk 4 above)
Comment (fliptopbox) [reply README.md:105]: yeah, it is and it doesn't show on the rendered MD. sure I'll refactor that.

File: README.md (hunk 3 above)
Comment (fliptopbox) [reply README.md:87]: what is the latest version?

File: README.md (hunk 2 above)
Comment (fliptopbox) [reply README.md:41]: (I'll delete that heading)

**Pull request merged**